Slim/Player/Protocols/HTTP.pm

```diff
@@ -320,10 +426,104 @@ sub sysread {
 			main::DEBUGLOG && $log->debug("The shoutcast metadata overshot the interval.");
 		}
 	}
+	
+	# when not-empty, chose return buffer length over sysread() 
+	$readLength = length $_[1] if length $_[1];
 
 	return $readLength;
 }
 
+sub _parseStreamHeader {
+	my ($self, undef, $readLength, $chunkSize) = @_;
+	my $args = ${*$self}{'parser_args'};	
+
+	return $readLength unless $args && $readLength;
+	
+	# stitch with trailing bytes
+	$_[1] .= $args->{'pending'};
```

Comment:
Please name this parameter instead of not initialising it at first.

Reply:
I'm confused by that comment

Reply:
It's another case of using a parameter by its anonymous name `$_[1]` instead of assigning it a local variable at the top of the sub.

Reply:
I think we can call this one resolve as well